test/unit/matrix/test_matrix.cpp

```diff
@@ -1833,6 +1834,63 @@ TYPED_TEST(MatrixTest, GPUCopySubPipeline) {
 }
 #endif
 
+struct TestReshuffling {
+  const GlobalElementSize size;
+  const TileElementSize src_tilesize;
+  const TileElementSize dst_tilesize;
+};
+std::vector<TestReshuffling> sizes_reshuffling_tests{
+    TestReshuffling{{10, 10}, {3, 3}, {3, 3}},   // no-reshuffling
```

Comment:
Do you agree in keeping this managed without any specific fallback logic to a simpler case?

Reply:
What do you mean?
As the `src_rank` is different the data has to be communicated anyway.